src/macaron/vsa/vsa.py

```diff
@@ -0,0 +1,248 @@
+    #: This field has the type
+    #: `ResourceURI <https://github.com/in-toto/attestation/blob/main/spec/v1/field_types.md#ResourceURI>`_.
+    #: Currently, this has the same value as the subject of the VSA, i.e. the PURL of
+    #: the software component being verified against.
+    resourceUri: str  # noqa: N815
+
+    #: The policy that the subject software component was verified against.
+    #: This field has the type
+    #: `ResourceDescriptor <https://github.com/in-toto/attestation/blob/main/spec/v1/resource_descriptor.md>`_.
+    policy: dict[str, Any]
+
+    #: The verification result.
+    verificationResult: VerificationResult  # noqa: N815
+
+    #: According to SLSA, this field "indicates the highest level of each track verified
+    #: for the artifact (and not its dependencies), or ``FAILED`` if policy verification failed".
+    #: We currently leave this list empty.
+    verifiedLevels: list  # noqa: N815
+
+
+class Verifier(TypedDict):
+    """The 'verifier' field within the Macaron VSA predicate field.
+
+    This field provides the identity of the verifier, as well as the versioning details of its components.
+    """
+
+    #: The identity of the verifier as a value of type
+    #:   `TypeURI <https://github.com/in-toto/attestation/blob/main/spec/v1/field_types.md#TypeURI>`_.
+    id: str  # noqa: A003
+
+    #: A mapping from components of the verifier and their corresponding versions.
+    #: At the moment, this field only includes Macaron itself.
+    version: dict[str, str]
+
+
+class VerificationResult(StrEnum):
+    """Verification result, which is either 'PASSED' or 'FAILED'."""
+
+    FAILED = "FAILED"
+    PASSED = "PASSED"
+
+
+def create_vsa_statement(
+    subject_purl: str,
+    policy_content: str,
+    verification_result: VerificationResult,
+) -> VsaStatement:
+    """Construct the Statement layer of the VSA.
+
+    Parameters
+    ----------
+    subject_purl : str
+        The PURL (string) of the subject of the VSA. This identifies the unique
+        software component that the policy applies to.
+    policy_content : str
+        The Souffle policy code defining the policy.
+    verification_result : VerificationResult
+        The verification result of the subject.
+
+    Returns
+    -------
+    VsaStatement
+        A Statement layer of the VSA.
+    """
+    return VsaStatement(
+        _type="https://in-toto.io/Statement/v1",
+        subject=[
+            {
+                "uri": subject_purl,
+            }
+        ],
+        predicateType="https://slsa.dev/verification_summary/v1",
+        predicate=VsaPredicate(
+            verifier=Verifier(
+                id="https://github.com/oracle/macaron",
+                version={
+                    "macaron": importlib_metadata.version("macaron"),
+                },
+            ),
+            timeVerified=datetime.utcnow().isoformat("T") + "Z",
```

Comment:
Instead of using concatenating concatenation like this, let's create an aware datetime object and call `isoformat()`.

Note that [the RFC allows](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6) `time-numoffset`, so we don't need to concatenate `Z`.

```suggestion
            timeVerified=datetime.now(tz=datetime.timezone.utc).isoformat()
```

Reply:
Thanks for the suggestion. See a59da72469db757674fb2dfe5af3851e8057c29c.

`pyupgrade` interestingly made the following change:

```diff
-            timeVerified=datetime.datetime.now(tz=datetime.timezone.utc).isoformat(),
+            timeVerified=datetime.datetime.now(tz=datetime.UTC).isoformat(),
```

Reply:
Oh nice. I didn't know the change in Python 3.11.